qcodes/tests/dataset/test_database_creation_and_upgrading.py

```diff
@@ -259,16 +260,275 @@ def test_perform_actual_upgrade_2_to_3_some_runs():
+        assert p5.depends_on == ''
+        assert p5.depends_on_ == []
+        # the 2 lines below are wrong due to the incorrect upgrade from
+        # db version 2 to 3. Here the interdep is missing
+        assert p5.inferred_from == ''
+        assert p5.inferred_from_ == []
+        assert p5.label == "Parameter 5"
+        assert p5.unit == "unit 5"
+
+        perform_db_upgrade_3_to_4(conn)
+
+        c = atomic_transaction(conn, sql)
+        json_str = one(c, 'run_description')
+
+        desc = RunDescriber.from_json(json_str)
+        idp = desc.interdeps
+        assert isinstance(idp, InterDependencies)
+
+        p0 = [p for p in idp.paramspecs if p.name == 'p0'][0]
+        assert p0.depends_on == ''
+        assert p0.depends_on_ == []
+        assert p0.inferred_from == ''
+        assert p0.inferred_from_ == []
+        assert p0.label == "Parameter 0"
+        assert p0.unit == "unit 0"
+
+        p1 = [p for p in idp.paramspecs if p.name == 'p1'][0]
+        assert p1.depends_on == ''
+        assert p1.depends_on_ == []
+        assert p1.inferred_from == ''
+        assert p1.inferred_from_ == []
+        assert p1.label == "Parameter 1"
+        assert p1.unit == "unit 1"
+
+        p2 = [p for p in idp.paramspecs if p.name == 'p2'][0]
+        assert p2.depends_on == ''
+        assert p2.depends_on_ == []
+        assert p2.inferred_from == 'p0'
+        assert p2.inferred_from_ == ['p0']
+        assert p2.label == "Parameter 2"
+        assert p2.unit == "unit 2"
+
+        p3 = [p for p in idp.paramspecs if p.name == 'p3'][0]
+        assert p3.depends_on == ''
+        assert p3.depends_on_ == []
+        assert p3.inferred_from == 'p1, p0'
+        assert p3.inferred_from_ == ['p1', 'p0']
+        assert p3.label == "Parameter 3"
+        assert p3.unit == "unit 3"
+
+        p4 = [p for p in idp.paramspecs if p.name == 'p4'][0]
+        assert p4.depends_on == 'p2, p3'
+        assert p4.depends_on_ == ['p2', 'p3']
+        assert p4.inferred_from == ''
+        assert p4.inferred_from_ == []
+        assert p4.label == "Parameter 4"
+        assert p4.unit == "unit 4"
+
+        p5 = [p for p in idp.paramspecs if p.name == 'p5'][0]
+        assert p5.depends_on == ''
+        assert p5.depends_on_ == []
+        assert p5.inferred_from == 'p0'
+        assert p5.inferred_from_ == ['p0']
+        assert p5.label == "Parameter 5"
+        assert p5.unit == "unit 5"
+
+
+def test_perform_upgrade_v3_to_v4_fixes():
+    """
+    Test that a db that was upgraded from v2 to v3 with a buggy
+    version will be corrected when upgraded to v4.
+    """
+
+    v3fixpath = os.path.join(fixturepath, 'db_files', 'version3')
+
+    dbname_old = os.path.join(v3fixpath, 'some_runs_upgraded_2.db')
+
+    if not os.path.exists(dbname_old):
+        pytest.skip("No db-file fixtures found. You can generate test db-files"
+                    " using the scripts in the legacy_DB_generation folder")
```

Comment:
The location of the scripts to generate fixtures has changed, so this skip message should reflect that change.